codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketServerProtocolHandler.java

```diff
@@ -154,7 +160,8 @@ public void handlerAdded(ChannelHandlerContext ctx) {
             // Add the WebSocketHandshakeHandler before this one.
             ctx.pipeline().addBefore(ctx.name(), WebSocketServerProtocolHandshakeHandler.class.getName(),
                     new WebSocketServerProtocolHandshakeHandler(websocketPath, subprotocols,
-                            allowExtensions, maxFramePayloadLength, allowMaskMismatch, checkStartsWith));
+                            allowExtensions, maxFramePayloadLength, allowMaskMismatch, checkStartsWith)
+                            .handshakeTimeoutMillis(handshakeTimeoutMillis));
```

Comment:
Ditto - no need to have a mutable state.

Reply:
Done!